### ISSUE TITLE
1 CPU: Visual Studio Code 100% CPU and System freeze on startup  win 7 64 bit

### ISSUE THREAD

**Issue by ParikshitBhagawati** [Microsoft/vscode#34396]
I have a windows 7 64bit PC running on an AMD processor with on board NVIDIA graphics chipset and 8 gb of RAM. I have tried various versions of vs code but the result is the same. It freezes the system for about 2 minutes until it opens it with a 100% CPU usage.

I have also tried different options like disable GPU and running without extentions etc. but all in vain.

I have an older laptop with 1gig RAM and windows 7 32 bit where vs code 1.1 runs without any of the above issues smoothly.

Can you please help ?

- VSCode Version: 1.1 tot 1.16
- OS Version: Windows 7 Ultimate 64bit

Steps to Reproduce:

1. Open visual studio code
2. System freeze and 100% CPU

Reproduces without extensions: Yes

**Comment by alexandrudima** [comment]
@ParikshitBhagawati VS Code uses multiple processes. Maybe it could help us figure out what's going on if we could find out which process is using 100% CPU. You can use a tool like process explorer and hover over the process that is at 100% CPU so we can find out the arguments it is launched with (i.e. identify which of our processes it is)

**Comment by ParikshitBhagawati** [comment]
@alexandrudima thanks for your response. attached is the screenshot from process explorer. please let me know if this help you with the info you were looking for. 
p.s - using the insiders build now.
![processexplorersnapshot](https://user-images.githubusercontent.com/11848537/30499473-442334b8-9a78-11e7-86f3-601f624bdd45.png)

**Comment by ParikshitBhagawati** [comment]
![perf-vscode_ii](https://user-images.githubusercontent.com/11848537/30511535-b2387dde-9af8-11e7-8dd0-f843e74448a9.png)

Here's more from today!

**Comment by alexandrudima** [comment]
I haven't seen so far our main process going crazy like this.

**Comment by bpasero** [comment]
Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: http://code.visualstudio.com/Download#insiders

**Comment by ParikshitBhagawati** [comment]
Well, the profile sent before was with the insiders build (Version 1.17.0). The link you provided however takes me to the download page for the latest stable build I guess, which is 1.16.1. But I downloaded it anyway with no change in results (100% CPU, unresponsive system etc.) and took 10 minutes to open.

Attached is the profile snapshot with the latest build.

![version-1 16 1](https://user-images.githubusercontent.com/11848537/30550135-3050b616-9cb4-11e7-86ff-a2ef25e72d65.png)

Thanks!

**Comment by bpasero** [comment]
@ParikshitBhagawati is this a single CPU VM?

**Comment by ParikshitBhagawati** [comment]
Its not a VM.. It's a native single core processor.

**Comment by bpasero** [comment]
Yeah. VS Code does not run well on 1-CPU systems. This is a known issue that was reported before.

**Comment by ParikshitBhagawati** [comment]
So is there a fix /work around for this yet or planned in the future ? Or it just wont work ?

**Comment by vaibhavsingh97** [comment]
I also facing same issue. After launching visual studio, RAM become 100% and my desktop freezes
OS: Arch Linux
Kernel: 4.13.4-1-ARCH

**Comment by ahumeniy** [comment]
In Windows you can launch VS Code with low or below normal CPU priority and it won't freeze your machine. If your're already affected you can open task manager and set the priority for the processes using 100% CPU to either low or below normal priority. After a while the high CPU usage will stop.

**Comment by royBrocken** [comment]
I changed my win7 VM to use multiple processors, and vscode is running fine again.

**Comment by lbertolotti** [comment]
My system is not freezing, but it almost did.

- VS code version 1.9.1 Architecture x64
- OS version:  Windows 10 Home
- Processor: Intel(R) Core(TM) i7-3630QM CPU @  2.40GHz 2.40 GHz
- Installed memory(RAM): 4.00 GB (3.89 GB usable)
- System type: 64-bit Operating System, x64-base processor
- memtest 86+ no errors found.

![firefox](https://user-images.githubusercontent.com/19675068/34441862-8ea1180a-eca5-11e7-862b-14334da197e2.png)

**Comment by ParikshitBhagawati** [comment]
As mentioned by @bpasero VS code doesn’t work well with single core CPUs. I had an option to enable multiple cores for my processor and when I did, Code started to work fine without the system freeze / 100% CPU issue.

**Comment by 9chu** [comment]
Meet the same issue on Windows Server 2012 (VM, 1 core cpu).

**Comment by sirhacksalot** [comment]
I no longer have this issue with version 1.21.

**Comment by johnsonz** [comment]
Same problem.

VS code: 1.20.1
OS: Win server 2012 R2

![capture](https://user-images.githubusercontent.com/2533629/38074115-41be7a30-3360-11e8-8ba0-c37b3976b8e1.JPG)

**Comment by ssteinerx** [comment]
Downloading an update in the background pegs the (a) CPU at 100%+ for me.  Always has, still does.

**Comment by Nhoya** [comment]
Same problem here, ArchLinux 4 core, latest vscode version

**Comment by philpalmieri** [comment]
Same here, i7 quad core with Ubuntu 18.04

**Comment by samwei12** [comment]
Same problem here, Windows10 12 core, latest vscode version

**Comment by mortenseifert** [comment]
I experience the same problem

    OS Name:                   Microsoft Windows Server 2012 R2 Standard
    OS Version:                6.3.9600 N/A Build 9600
    OS Manufacturer:           Microsoft Corporation
    OS Configuration:          Member Server
    OS Build Type:             Multiprocessor Free
    Registered Owner:          Windows User
    Registered Organization:
    Product ID:                00252-60272-42486-AA665
    Original Install Date:     24-10-2017, 09:35:44
    System Boot Time:          02-05-2018, 14:30:53
    System Manufacturer:       VMware, Inc.
    System Model:              VMware Virtual Platform
    System Type:               x64-based PC
    Processor(s):              1 Processor(s) Installed.

**Comment by thanhdevapp** [comment]
I experience the same problem

**Comment by scasic** [comment]
same here,  linux mint 18.1
when I close it VSCode its process still remains in processes list (100%)

**Comment by DFineNormal** [comment]
Same problem here. Workaround for now appears to be to reduce Code.exe process priority to Below Normal, otherwise it's a good 10 mins to open and various actions cause it to freeze again after that.

Same issue on multiple Windows Server 2008 R2 SP1 VMs, all with dual vCPUs and various versions of VS code, currently 1.23.1.

**Comment by pavelloz** [comment]
Problem also happens on OSX 10.12.6

**Comment by fasterthan** [comment]
@bpasero I am having the same issue on an AMD Ryzen 1300x (four physical cores) so I don't think it is a single core issue.

![vscode_high_cpu_usage](https://user-images.githubusercontent.com/13704150/42406928-bf80a160-8166-11e8-808e-0ed39a2bdfda.png)

**Comment by hiwjcn** [comment]
same here

**Comment by JohnProg** [comment]
same here

**Comment by jannikbuhr** [comment]
same issue on Win 10, with code and code-exploration. Also code doesn't open without internet connection

**Comment by lyltje** [comment]
same here, it makes the IDE almost unusable

**Comment by r4hulCorleone** [comment]
I had same problem with newest update ,1 physical core ,100 cpu usage,window 7
[Solution]:change compability mode to windows xp sp3 for vs code solved my problem.

**Comment by ergenekonyigit** [comment]
same problem here. 

- macOS 10.14.1 
- code 1.29.1

**Comment by CookieCrumbles** [comment]
Same problem here. I reverted back to 1.25
Seems like that is the latest usable version for me. 
**Is there any update on a fix ????**

**Comment by gustaff-weldon** [comment]
Experiencing the same issue. CodeHelper is stuck on ~100%.
<img width="561" alt="screen shot 2018-11-21 at 09 53 09" src="https://user-images.githubusercontent.com/411756/48829732-4c2f7280-ed73-11e8-9551-67878c4d18e2.png">

`ps aux` output on that process:
```
✗ ps aux | grep 13158
-edited-    13158  99.5  1.4  6194332 242864   ??  R    Tue09AM  59:28.99 /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js --useInferredProjectPerProjectRoot --enableTelemetry --cancellationPipeName /var/folders/5b/gz8j6f890kb2pd1n6zfn09c00000gn/T/vscode-typescript/tscancellation-87c31d99c306e6ae821e.tmp* --locale en --noGetErrOnBackgroundUpdate
```

Looks like `Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js` is a culprint. My project is a monorepo with a lot of JS code (not TS enabled)

**Comment by lautis0503** [comment]
This could be a bug from electron. There's a bug fix finally.
https://github.com/electron/electron/pull/15975

**Comment by bpasero** [comment]
I finished a build with the fix from electron/electron#15975:

https://az764295.vo.msecnd.net/insider/5967828c2a5c559ed86c3369ed9811bb1bd6386b/VSCode-win32-x64-1.31.0-insider.zip

Can people report back if this is fixed with that build?

**Comment by 9chu** [comment]
> I finished a build with the fix from [electron/electron#15975](https://github.com/electron/electron/pull/15975):
> 
> https://az764295.vo.msecnd.net/insider/5967828c2a5c559ed86c3369ed9811bb1bd6386b/VSCode-win32-x64-1.31.0-insider.zip
> 
> Can people report back if this is fixed with that build?

![image](https://user-images.githubusercontent.com/4137042/50688349-064b1700-1060-11e9-9e7f-a5afe46c8002.png)

Solved on my VPS with 1 core CPU, running Windows Server 2012 R2.

**Comment by DFineNormal** [comment]
> I finished a build with the fix from [electron/electron#15975](https://github.com/electron/electron/pull/15975):
> 
> https://az764295.vo.msecnd.net/insider/5967828c2a5c559ed86c3369ed9811bb1bd6386b/VSCode-win32-x64-1.31.0-insider.zip
> 
> Can people report back if this is fixed with that build?

This build fixed the issue for me.
Tested on 2 x single core VMs running Win 2008 and 2008 R2 respectively

**Comment by bpasero** [comment]
Very cool, thanks for confirming!

**Comment by bpasero** [comment]
Closing this issue given that we plan to release VSCode stable early February with Electron 3.x. If you want to benefit from the fix already, consider to use our insiders version that already contains the fix: https://code.visualstudio.com/insiders/

**Comment by CookieCrumbles** [comment]
Insider or not; electron 3 or not, VS Code takes minutes to just open up. 
My only good working version is v1.25. Anything above that is so slow that it becomes unusable at this point. There are so many people mentioning this but i haven't read about a fix. Yeah you can have a look into the processes but that gives you numbers we don't even understand. I really wish you guys either came out with a fix OR with some better way for us to help finding out what is causing this.

**Comment by viniciusgati** [comment]
> > I finished a build with the fix from [electron/electron#15975](https://github.com/electron/electron/pull/15975):
> > https://az764295.vo.msecnd.net/insider/5967828c2a5c559ed86c3369ed9811bb1bd6386b/VSCode-win32-x64-1.31.0-insider.zip
> > Can people report back if this is fixed with that build?
> 
> This build fixed the issue for me.
> Tested on 2 x single core VMs running Win 2008 and 2008 R2 respectively

Me to.

**Comment by ssj7** [comment]
I have a core i7 and it still takes about 10 sec to open an ionic project. I have 16GB ram.
